assert-struct-macros/src/expand.rs

```diff
@@ -1028,30 +1061,39 @@ fn generate_plain_tuple_assertion(
     elements: &[Pattern],
     is_ref: bool,
 ) -> TokenStream {
-    // Generate unique names to avoid conflicts
-    let element_names: Vec<_> = (0..elements.len())
-        .map(|i| quote::format_ident!("__tuple_elem_{}", i))
-        .collect();
+    // Generate match patterns and bindings for elements
+    let mut match_patterns = Vec::new();
+    let mut element_assertions = Vec::new();
+
+    for (i, pattern) in elements.iter().enumerate() {
```

Comment:
Ok, this is the third time I've seen a snippet that looks almost identical. DRY up this code.